terraform/s3/provision/data.tf

```diff
@@ -0,0 +1,8 @@
+locals {
+  ol_configuration_has_retention_mode  = var.ol_configuration_default_retention_mode != null
+  ol_configuration_has_retention_days  = var.ol_configuration_default_retention_days != null
+  ol_configuration_has_retention_years = var.ol_configuration_default_retention_years != null
+  ol_configuration_has_retention       = (local.ol_configuration_has_retention_mode || local.ol_configuration_has_retention_days || local.ol_configuration_has_retention_years)
+  ol_configuration_is_enabled          = var.ol_configuration_default_retention_enabled != null ? var.ol_configuration_default_retention_enabled : local.ol_configuration_has_retention
+  is_versioning_enabled                = var.enable_versioning ? true : var.ol_enabled
```

Comment:
Am I right in thinking this could evaluate to `true` or `true` in some cases? I may not fully grasp the logic here, but it seems a little confusing to me to have a ternary operator where both values could be the same.